sol/Week2/useContext/src/context/ThemeToggleButton.tsx

```diff
@@ -0,0 +1,19 @@
+import { THEME, useTheme } from "./ThemeProvider";
+import clsx from "clsx";
+export default function ThemeToggleButton() {
+  const { theme, toggleTheme } = useTheme();
+
+  const isLightMode = theme === THEME.LIGHT;
+
+  return (
+    <button
+      onClick={toggleTheme}
+      className={clsx("px-4 py-2 mt-4 rounded-md transition-all", {
+        "bg-black text-white": !isLightMode,
+        "bg-white text-black": isLightMode,
+      })}
```

Comment:
질문 주신 clsx는 css에서 여러 클래스 객체를 조건부로 쓰려고 할 때 사용합니다! 여기서두 isLightMode의 값에 따라서 적용하는 css를 다르게 하고 있는 걸 볼 수 있죠~!
사실 clsx는 import가 필요하기도 하고, clsx를 쓰는 것보다 그냥 리터럴로 쓰는 게 간편하기도 해서 리터럴로 쓰는 경우가 더 많은 것 같아요:)

Reply:
리터럴 예를 들면 요론 식으로 별도 라이브러리 필요없이 조건부 처리를 할 수 있답니다~!!
```suggestion
className={`px-4 py-2 mt-4 rounded-md transition-all ${
  isLightMode ? "bg-white text-black" : "bg-black text-white"
}`}
```

Reply:
혹시 혼동하시나 싶어 말씀드리면, tailwindcss는 css를 간편하게 쓰기 위한 css 프레임워크이고, clsx는 css를 간편하게 쓰기 위한 하나의 함수? 정도로 생각해주시면 될 것 같아요. 테일윈드가 훨씬 넓은 개념이에요!

Reply:
정말 감사합니다!!!!!